tests/Microsoft.DotNet.Framework.Docker.Tests/GeneratedArtifactTests.cs

```diff
@@ -0,0 +1,55 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.ComponentModel;
+using System.Diagnostics;
+using System.IO;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.DotNet.Framework.Docker.Tests
+{
+    [Trait("Category", "pre-build")]
+    public class GeneratedArtifactTests
+    {
+        private ITestOutputHelper OutputHelper { get; }
+
+        public GeneratedArtifactTests(ITestOutputHelper outputHelper)
+        {
+            OutputHelper = outputHelper;
+        }
+
+        [Fact]
+        public void VerifyReadmeTemplates()
+        {
+            string generateTagsDocumentationScript = Path.Combine(Config.SourceRepoRoot, "eng", "readme-templates", "Get-GeneratedReadmes.ps1");
+            ValidateGeneratedArtifacts(
+                generateTagsDocumentationScript,
+                $"The Readmes are out of sync with the templates.  Update the Readmes by running `{generateTagsDocumentationScript}`.");
+        }
+
+        private void ValidateGeneratedArtifacts(string generateScriptPath, string errorMessage)
+        {
+            string powershellArgs = $"-File {generateScriptPath} -Validate";
+            (Process Process, string StdOut, string StdErr) executeResult;
+
+            // Support both execution within Windows 10, Nano Server and Linux environments.
```

Comment:
Doesn't need to account for Nano Server and Linux in this repo.

Reply:
It does actually need to support Linux.  The pre-build validation is run on a Linux build agent for PR validation and official builds.  When ltsc2016 no longer needs to be supported then we will be able to run the Windows legs within a Docker image which may be Nano Server based.  Because of this, I am going to leave the comment as is.